### PR TITLE
Nicholas Connors

### PR DESCRIPTION
**Pull request by NicholasConnors**
Assignment 4 + bonus challenge

### CODE REVIEW

**Comment by frank-ye** [comment]
Check Slack for the notebook with theoretical answers, but yours are well written, good job! Good work on Latex equations as well.
2) Good job on the plotting for SVM question, simple yet effective visualization
3) These data distributions weren't inherently taught in class, good stuff! 
4a) See slack for answers (However, your pictures are not appearing)
4b) Good
5) See slack for answers (However, your pictures are not appearing)
6) See slack for answers
7) Weird that your dimensionality reduction doesn't increase accuracy, probably an error in implementation
You did the bonus challenge, congrats this is noted. Good idea applying NN to the PCA reduced features, just being nitpicky but you definitely overfit the neural network, acc_val <<< acc_train
Grade - Pass
Great job, you should apply to ML internships
